frontend/packages/kubevirt-plugin/src/components/modals/boot-order-modal/boot-order-modal.tsx

```diff
@@ -0,0 +1,156 @@
+  }, [vmLikeEntity]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Inform user on vmLikeEntity.
+  React.useEffect(() => {
+    // Check if bootOrder of current device changed.
+    if (!_.isEqualWith(initialDeviceList, getDevices(vmLikeEntity), _.isEqual)) {
+      setUpdatedAlert(true);
+    }
+  }, [vmLikeEntity]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Re-set device list on isOpen change to true.
+  React.useEffect(() => {
+    if (isOpen) {
+      onReload();
+    }
+  }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Send new bootOrder to k8s.
+  const onSubmit = async (event) => {
+    event.preventDefault();
+
+    // Copy only bootOrder from devices to current device list.
+    const currentDevices = _.cloneDeepWith(getDevices(vmLikeEntity));
+    devices.forEach((d) => {
+      // Find the device to update.
+      const device = currentDevices.find(
+        (cd) => d.type === cd.type && d.value.name === cd.value.name,
+      );
+
+      // Update device bootOrder.
+      if (device && d.value.bootOrder) {
+        device.value.bootOrder = d.value.bootOrder;
+      }
+      if (device && device.value.bootOrder && !d.value.bootOrder) {
+        delete device.value.bootOrder;
+      }
+    });
+
+    // Filter disks and interfaces from devices list.
+    const disks = [
+      ..._.filter(currentDevices, (source) => source.type === DeviceType.DISK).map(
+        (source) => source.value,
+      ),
+    ];
+    const interfaces = [
+      ..._.filter(currentDevices, (source) => source.type === DeviceType.NIC).map(
+        (source) => source.value,
+      ),
+    ];
+
+    // Patch k8s.
+    const patches = [
+      new PatchBuilder('/spec/template/spec/domain/devices/disks')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(disks)
+        .build(),
+      new PatchBuilder('/spec/template/spec/domain/devices/interfaces')
+        .setOperation(PatchOperation.REPLACE)
+        .setValue(interfaces)
+        .build(),
+    ];
+    const promise = k8sPatch(
+      getVMLikeModel(vmLikeEntity),
+      vmLikeEntity,
+      getVMLikePatches(vmLikeEntity, () => patches),
+    );
+
+    promise
+      .then(() => setOpen(false))
+      .catch((e: any) => setPatchError(e ? e.toString() : 'Patch failed.'));
+  };
+
+  const actions = [
+    <Button key="confirm" variant="primary" onClick={onSubmit}>
+      Save
+    </Button>,
+    <Button key="cancel" variant="link" onClick={() => setOpen(false)}>
+      Cancel
+    </Button>,
+  ];
+
+  return (
+    <Modal
+      title={title}
+      isOpen={isOpen}
+      isSmall
+      onClose={() => setOpen(false)}
+      actions={actions}
+      isFooterLeftAligned
+    >
+      <BootOrder devices={devices} setDevices={setDevices} />
+
+      <div className="kubevirt-boot-order-modal__alerts">
+        {showUpdatedAlert && (
+          <Alert variant="warning" isInline title="Boot order has been updated outside this flow.">
```

Comment:
This isn't always true, as we detect all changes. E.g. just bus could have changed, while boot order stays untouched.

Reply:
now trigger only on boot order changes